tests/core.py

```diff
@@ -1865,6 +1865,31 @@ def tearDown(self):
         configuration.conf.remove_option("core", "SECURE_MODE")
 
 
+class PasswordUserTest(unittest.TestCase):
+    def setUp(self):
+        from airflow.contrib.auth.backends.password_auth import PasswordUser
+        user = models.User()
+        self.password_user = PasswordUser(user)
+        self.password_user.username = "password_test"
+
+    @mock.patch('airflow.contrib.auth.backends.password_auth.generate_password_hash')
+    def test_password_setter(self, mock_gen_pass_hash):
+        mock_gen_pass_hash.return_value = b"hashed_pass" if six.PY3 else "hashed_pass"
+
+        self.password_user.password = "secure_password"
+        mock_gen_pass_hash.assert_called_with("secure_password", 12)
+
+    def test_password_unicode(self):
+        # In python2.7 no conversion is required back to str
+        # In python >= 3 the method must convert from bytes to str
+        self.password_user.password = "secure_password"
+        self.assertIsInstance(self.password_user.password, str)
+
+    def test_password_user_authenticate(self):
+        self.password_user.password = "secure_password"
+        self.assertTrue(self.password_user.authenticate("secure_password"))
```

Comment:
Could you add one more test - like this one, but that round-trips through the database.

Reply:
Sure, just a little clarification, is it any different than these tests?
https://github.com/apache/incubator-airflow/blob/master/tests/core.py#L1868-L1932

Reply:
Sort of? Those tests are testing the auth of the webserver. I am happy with a separate test of the PasswordUser model (as you have here) or of them being merged into the same TestCase.

Though I am somewhat wondering why those tests weren't failing on master.... curious.